README.md

```diff
@@ -32,47 +32,52 @@ A hotel booking application in React. Homework for the [CodeYourFuture React mod
+20. Now instead of getting the existing bookings from the file `data/fakeBookings.json`, we will get the bookings from a remote API. In the `<Bookings />` component, use the React function `useEffect` to console log some text only when the page first renders on the screen. Verify that when you refresh the page, the text appears once in the console. Then, in the `useEffect` function, use the `fetch()` function to get data from `https://cyf-react.glitch.me`. Hints:
 
-20. Again in the `<Bookings />` component, use the `fetch()` function to get data from `https://cyf-react.glitch.me`. Hints:
+- Replace `FakeBookings` in the bookings state and initialise it with `[]` (because we haven't fetched any results yet!)
+- After calling the `fetch()` function, use `.then()` to handle the response. Try looking at your Pokemon app that you worked on in class for an example
+- When the response comes back, use `setBookings` to update the results
 
-- Replace `FakeBookings` in the state initialise with `null` (because we haven't fetched any results yet!)
-- Add a `componentDidMount()` method that calls the `fetch()` function and then use `.then()` to handle the response. Try looking at your Pokemon app that you worked on in class for an example
-- When the response comes back use `this.setState` to update the results
+Verify the customers data are still displayed correctly in the table.
 
-21. Now show a _loading state_ in `<Bookings />` while the data from the server is being fetched. To test this, try loading data from `https://cyf-react.glitch.me/delayed`, which has a 5 second delay before returning the data. Hint: try looking at your Pokemon app that you worked on in class for an example
+21. Now show a _loading state_ in `<Bookings />` while the data from the server is being fetched. To test this, try loading data from `https://cyf-react.glitch.me/delayed`, which has a 5 second delay before returning the data. You will need to use another state to record when your application is loading data (this can be a boolean) and display a loading message whenever the application is loading data. Hint: try looking at your Pokemon app that you worked on in class for an example.
 
-22. Finally, display an error message in `<Bookings />` if there is an HTTP error when fetching data from the server. To test this, try loading data from `https://cyf-react.glitch.me/error`, which will return a 500 HTTP error. Hint: Try looking at your Pokemon app that you worked on in class for an example
+22. Finally, display an error message in `<Bookings />` if there is an HTTP error when fetching data from the server. To test this, try loading data from `https://cyf-react.glitch.me/error`, which will return a 500 HTTP error. Hint: Try looking at your Pokemon app that you worked on in class for an example.
```

Comment:
Do we need to provide information for `.catch()` when the fetch fails or should students be comfortable with this?

Reply:
I would have to double check with what was taught in JS2/3 but I believe they would be familiar with `.catch()`. Given that this is the last exercise, I think leaving off some hints is likely fine.